src/java/main/org/apache/zookeeper/server/FinalRequestProcessor.java

```diff
@@ -476,6 +624,33 @@ private boolean connClosedByClient(Request request) {
         return request.cnxn == null;
     }
 
+    private String getHostAddress(ServerCnxn cnxn) {
```

Comment:
perhaps we could do this logic in ServerCnxn?

Reply:
moved